### DISCUSSION TITLE
Notes Disappeared

### DISCUSSION THREAD

**Discussion by segundaestrella** [alagrede/znote-app#23] [Q&A] [answered]
Guys i need help, all my notes disappeared after an os startup. I have installed some os updates to some apps (not znote between them) and then i restarted the laptop. Once restarted all my notes were gone.
Please I need a way to recover them, there were important infos saved there.

**Comment by alagrede** [comment, marked as answer]
Hello @segundaestrella,
Did you install Znote via Store or directly the preview lab? Win/Mac?
If you've created a cloud account and sync your notes, you can restore them just with a login. (The cloud sync concerns only notes, your physical files should still exist on your file system)

Otherwise, you can check if your data is still present in the **user-preferences.json** file

The file can have different locations:
For the official Store version
`~/Library/Containers/com.tony.znote/Data/Library/Application\ Support/znote`
Or for the download via the website (Lab version)
`~/Library/Application\ Support/znote`
Under Windows:
`~\AppData\Roaming\znote\`

I hope you will be able to find your data 🤞

**Reply by segundaestrella** [reply]
Thank you very much for your quick answer, I downloaded it from the website (lab version), but it seems it doesn't have created that folder in Application Support, and neither I have the Containers folder (I was checking just in case)

![image](https://user-images.githubusercontent.com/19385900/121928004-9f091500-cd37-11eb-9948-3489b4294db2.png)

**Reply by alagrede** [reply]
Check your user data: `/Users/segundaestrella/Library/Application\ Support/znote`

**Comment by segundaestrella** [comment]
I don't have neither that folder

![image](https://user-images.githubusercontent.com/19385900/121928891-877e5c00-cd38-11eb-9adf-74f064e255ec.png)

**Reply by alagrede** [reply]
It's an hidden folder. Try to open the **terminal** and copy/paste the command below:
`open ~/Library/Application\ Support/znote`

**Reply by segundaestrella** [reply]
There is the user-preferences.json file! how do i import back?

**Reply by alagrede** [reply]
Try to open the file with another editor and tell me if you see your notes inside?

**Reply by segundaestrella** [reply]
yes i do!

**Reply by alagrede** [reply]
Ok if you find your data, **make a backup** of this file to your Desktop to be sure to keep a copy.
Maybe there is a bug or a malformed json content into the file. You can try to tweak the file to retrieve a valid state.
If your data are not too sensible, you can send me the user-preference.json file by mail: **lagrede.anthony[at]gmail.com** I could made the analyse for you and try to repair the file.

**Reply by segundaestrella** [reply]
No is a valid JSON i can see it with the formatter and linting, in any case i can fix it if something broken. Can you just explain me the procedure to re-import them?

**Reply by alagrede** [reply]
Theoretically, the file is played automatically when you launch Znote. If the list of notes is empty and contains only "default notes", maybe the *user-preferences.json* file is empty: s
1. Make a backup of user-preferences.json
2. Update to the last Znote version: https://znote.io/lab.html

If your Znote doesn't detect the file. You have 2 options:
- Try to remove json notes manually into user-preferences.json file and restart Znote until it works.
- Or delete the user-preferences.json, Znote will create a fresh empty file and copy manually the content of your all notes from the old backup file

In all cases, I will be interested to have a cleaned copy of your file to better understand what happened. So I could correct this failure.

**Reply by segundaestrella** [reply]
Ok Im going to try as you suggest.
Here the screenshot with the znote instance, the finder and the file infos.
Im also going to send you the file by mail, please let me know if you discover something!
Thanks a lot for the support

![image](https://user-images.githubusercontent.com/19385900/121934580-610fef00-cd3f-11eb-8642-5ab909229dd9.png)

**Reply by alagrede** [reply]
I just tried your file, it works fine! Just try to update the app!

**Reply by segundaestrella** [reply]
Thank you very much, your help is been incredible!
As last point can i ask you to tell me where i can find the documentation about storing the notes on icloud or google drive?

**Reply by alagrede** [reply]
I am really happy that you did not lose your data! 😅 
I completely deleted your mail and json file. 

Check out this blog post to find out what new features you missed: https://blog.znote.io/2021/labfeatures-post/

About the cloud account, I haven't write any documentation.
Create a free account (click on the bottom right cloud icon)
<img width="904" alt="create-account" src="https://user-images.githubusercontent.com/5312754/121947087-f0c09800-cd55-11eb-84f7-55632aae7a6d.png">

And make sure to turn on sync for every note you want to sync:
<img width="241" alt="cloud" src="https://user-images.githubusercontent.com/5312754/119521206-37d0f400-bd7b-11eb-93ed-ddaa8c3a2271.png">


Cheers
